allennlp/nn/util.py

```diff
@@ -622,10 +624,10 @@ def get_text_field_mask(
     smallest_dim = tensor_dims[0][0] - num_wrapping_dims
     if smallest_dim == 2:
         token_tensor = tensor_dims[0][1]
-        return token_tensor != 0
+        return token_tensor != padding_id
     elif smallest_dim == 3:
         character_tensor = tensor_dims[0][1]
-        return (character_tensor > 0).any(dim=-1)
+        return (character_tensor != padding_id).any(dim=-1)
```

Comment:
Do you mean specifically the change from > 0 to != or just the addition of a custom padding id in general? I think the two are equivalent since indices should never be negative.

I added two tests for using the custom padding id.

Reply:
Thanks!